concordium-consensus/src/Concordium/Scheduler/Environment.hs

```diff
@@ -498,6 +532,8 @@ data LocalState = LocalState{
   _energyLeft :: !Energy,
   -- |Changes accumulated thus far.
   _changeSet :: !ChangeSet,
+  -- |Maximum number of modified contract instances.
+  _nextContractModificationIndex :: !ModificationIndex,
```

Comment:
I already added some, but the precise semantics of this will change with the new state branch, so I will add more details there (and in the file documenting the semantics of V1 contracts).